bionic/code_analyzer.py

```diff
@@ -0,0 +1,134 @@
+"""
+This module implements the logic to analyze your code objects and
```

Comment:
Can we give a higher-level description of what's happening here? Why do we need to analyze code objects?